docs/build/js-client/10-marine-js.md

```diff
@@ -1,63 +1,79 @@
 
-# TODO: adapt to JS Client
+## Using Marine services in JS client
 
-## Using Marine services in Fluence JS
+JS client can host Marine services with Marine JS. Currently only pure single-module services are supported.
 
-Fluence JS can host Marine services with Marine JS. Currently only pure single-module services are supported.
+Before registering the service, the corresponding WASM file must be loaded.
 
-Before registering the service, the corresponding WASM file must be loaded. Fluence JS package exports three helper functions for that.
+Here are the examples of how to load your wasm file in browser or node
 
-**loadWasmFromFileSystem**
+**Node.JS**
 
-Loads the WASM file from the file system. It accepts the path to the file and returns a buffer compatible with the `FluencePeer` API.
+```javascript
+import { readFile } from 'fs/promises';
 
-This function can only be used in nodejs. Trying to call it inside a browser will throw an error.
-
-**loadWasmFromNpmPackage**
-
-Locates a WASM file in the specified npm package and loads it. The function accepts two arguments:
-
-- Name of an npm package
-- Path to a WASM file relative to the npm package
-
-This function can only be used in nodejs. Trying to call it inside a browser will result in an error.
+const wasm = await readFile('path/to/wasm/file.wasm', 'base64');
+```
 
-**loadWasmFromServer**
+**Browser**
 
-Loads a WASM file from the service hosting the application. It accepts the file path on server and returns a buffer compatible with the `FluencePeer` API.
+> Need to install additional external package in browser env - `js-base64`
 
-:::info
-The function will try to load a file into SharedArrayBuffer if the site is cross-origin isolated. Otherwise, the return value falls back to Buffer, which is less performant but is still compatible with the `FluencePeer` API.
+```javascript
+import { fromUint8Array } from 'js-base64';
 
-We strongly recommend to set-up cross-origin headers. For more details, see the [MDN page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer#security_requirements).
-:::
+const wasmBinary = await fetch('https://wasm.com/url/to/wasm').then(res => res.arrayBuffer());
+const wasm = fromUint8Array(new Uint8Array(greetingWasm));
+```
 
-This function can only be used in a browser. Trying to call it inside nodejs will result in an error.
 
 ### Registering services in FluencePeer
 
-After the file has been loaded, it can be registered in `FluencePeer`. To do so, use the `registerMarineService` function.
-
-To remove a service, use the `registerMarineService` function.
+JS client loads wasm content through its own special service called `Srv`.
+You need to add the service definition if you want to use it. 
 
-You can pick any unique service id. Once the service has been registered it can be referred in Aqua code by the specified id. For example:
+Add the following aqua file in your project near the other aqua files.
```

Comment:
Do you need to do that? Isn't thata part of aqua-lib?

Reply:
Yes, aqua-lib has inconsistent definition with JS-client API. Currently there is a specific feature of client

Reply:
why is it inconsistent. Are there any plans to make it consistent? Or maybe a separate definiton is needed in aqua-lib specifically for what js-client can do. Because asking people to add a file like that doesn't look good in my opinion. You can ask e.g. somebody elses opinion on that as well

Reply:
You're right. This is very experimental and hacky way to utilize module registration. Ideally we need to make it consistent with aqua-lib.
I made a [task](https://linear.app/fluence/issue/DXJ-563/make-marine-service-registration-consistent-with-aqua-lib) for this.

Reply:
But doing this now is costly. 
I will add 'experimental' label to the page.